src/fields/fp.rs

```diff
@@ -580,11 +563,30 @@ impl ff::Field for Fp {
 impl ff::PrimeField for Fp {
     type Repr = [u8; 32];
 
+    const MODULUS: &'static str =
+        "0x40000000000000000000000000000000224698fc094cf91b992d30ed00000001";
+    const TWO_INV: Self = Fp::from_raw([
+        0xcc96987680000001,
+        0x11234c7e04a67c8d,
+        0x0000000000000000,
+        0x2000000000000000,
+    ]);
```

Comment:
I have not checked any of the constants.

Reply:
The constants should all be pure moves; no news constants were calculated here.